### PR TITLE
Curriculum Improvements: Grammar, Styling, and Resources for Weeks 7, 8, & 11 (#2116)

### PR DESCRIPTION
**Pull request by mai-repo**
### Summary:
This PR includes various improvements across multiple projects, focusing on adding resources, fixing grammatical issues, and enhancing documentation. The following files were audited:

- [Weather App](https://github.com/Techtonica/curriculum/blob/main/projects/weather-app.md) (i.e. weather-app.md)
- [Quiz App](https://github.com/Techtonica/curriculum/blob/main/projects/week8GameREADME.md) (i.e. week8GameREADME.md)
- [Contact App](https://github.com/Techtonica/curriculum/blob/main/projects/pern-contact-list-app.md) (i.e. pern-contact-list-app.md)
- [Weather2](https://github.com/Techtonica/curriculum/blob/main/projects/pern-weather-app.md) (i.e. pern-weather-app.md) -or- [Game2](https://github.com/Techtonica/curriculum/blob/main/projects/pern-game-app.md) (i.e. pern-game-app.md)

### Changes Made:

- Added resource for saving API key in .env file in `weather-app.md` 
- Added Stack Overflow thread resource for decoding strings with special HTML entities in the `week8GameREADME.md` file 
- Added checklist for required features in README and fixed minor grammatical errors in the file `pern-contact-list-app.md`
- Fixed grammatical errors and styling in `pern-weather-app.md` 
- Fixed grammatical errors and styling in `pern-game-app.md`

### Objective:
The purpose of this ticket is to audit, report, and implement needed changes to the curriculum for areas of the existing full time program projects that aren't clear.

### Next Steps:
- Reviewers: @daaimah123, @mx-ruthie, and @ChasVanDav please see the following changes to the files: 

1. `weather-app.md` 
2. `week8GameREADME.md` 
3. `pern-contact-list-app.md`
4. `pern-weather-app.md` 
5. `pern-game-app.md`

### CODE REVIEW

**Review by daaimah123 — APPROVED** [review]

**Review by ChasVanDav — APPROVED** [review]
Great job keeping formatting consistent with capitalization, punctuation, etc. and adding quotes around file names for clarity!

File: projects/weather-app.md [hunk 1]
```diff
@@ -96,8 +97,8 @@ When you press the submit button on the form you will get data from the API. Mak
 
 ## Guide Code
 
-You can see a guide code working with hard-code data [here](https://github.com/Yosolita1978/HardCodeDataWeatherApi)
-You can see a guide code working with real data with a API_KEY [here](https://github.com/Yosolita1978/RealDataWeatherAPI)
+- You can see a guide code working with hard-code data [here](https://github.com/Yosolita1978/HardCodeDataWeatherApi).
```
Comment (mx-ruthie) [inline projects/weather-app.md:100]: You can see an example of working with hard-coded data [here]

File: projects/weather-app.md [hunk 2]
```diff
@@ -96,8 +97,8 @@ When you press the submit button on the form you will get data from the API. Mak
...
+- You can see a guide code working with real data with a API_KEY [here](https://github.com/Yosolita1978/RealDataWeatherAPI).
```
Comment (mx-ruthie) [inline projects/weather-app.md:101]: You can see an example of working with real data coming from the use of an API_KEY [here]

**Review by mx-ruthie — APPROVED** [review]
Looks good to me. I added a few grammatical changes in one spot but otherwise :shipit:

File: projects/weather-app.md (hunk 1 above)
Comment (mai-repo) [reply projects/weather-app.md]: Thanks, @mx-ruthie! Just committed that change and the commit # is d593b86152fb5b294a28c548606525bd937c7113

File: projects/weather-app.md (hunk 2 above)
Comment (mai-repo) [reply projects/weather-app.md]: Thanks for pointing this out! Just committed the suggested change, the commit # for that is 7f00590e60a03023c0b312567f065126b918a4a8.

**Comment by mai-repo** [comment]
@daaimah123 I see that you've approved the PR. Are there any additional changes you'd like me to make? If not, it looks like I have all the approvals needed to proceed with the merge. :)

**Pull request merged**